homeassistant/components/media_player/anthemav.py

```diff
@@ -0,0 +1,173 @@
+"""
+Support for Anthem Network Receivers and Processors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.anthemav/
+"""
+import logging
+import asyncio
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_SELECT_SOURCE,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, STATE_OFF, STATE_ON, STATE_UNKNOWN,
+    EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['anthemav==1.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'anthemav'
+
+DEFAULT_PORT = 14999
+
+SUPPORT_ANTHEMAV = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    })
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up our socket to the AVR."""
+    import anthemav
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    name = config.get(CONF_NAME)
+
+    _LOGGER.info('Provisioning Anthem AVR device at %s:%d', host, port)
+
+    def anthemav_update_callback(message):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.info('Received update calback from AVR: %s', message)
+        device.schedule_update_ha_state()
```

Comment:
You have a asyncio library and call all stuff inside Loop. This function can only called from outside the Loop. Use `hass.loop.create_task(device.async_update_ha_state())` but you need sure that you understund asyncio and run all stuff inside Loop!